packages/jest-runtime/src/index.js

```diff
@@ -496,6 +496,61 @@ class Runtime {
     return to ? this._resolver.resolveModule(from, to) : from;
   }
 
+  _requireResolve(
+    from: Path,
+    moduleName?: string,
+    {paths}: {paths?: Path[]} = {},
+  ) {
+    if (moduleName == null) {
+      throw new Error(
+        'The first argument to require.resolve must be a string. Received null or undefined.',
+      );
+    }
+
+    if (paths) {
+      for (const p of paths) {
+        const absolutePath = path.resolve(from, '..', p);
+        const module = this._resolver._resolveModuleFromDirIfExists(
+          absolutePath,
+          moduleName,
+          // required to also resolve files without leading './' directly in the path
+          {paths: [absolutePath]},
+        );
+        if (module) {
+          return module;
+        }
+      }
+      throw new Error(
+        `Cannot resolve module '${moduleName}' from paths ['${paths.join(
+          "', '",
+        )}'] from ${from}`,
+      );
+    }
+
+    return this._resolveModule(from, moduleName);
+  }
+
+  _requireResolvePaths(from: Path, moduleName?: string) {
+    if (moduleName == null) {
+      throw new Error(
+        'The first argument to require.resolve.paths must be a string. Received null or undefined.',
+      );
+    }
+    if (!moduleName.length) {
+      throw new Error(
+        'The first argument to require.resolve.paths must not be the empty string.',
+      );
+    }
+
+    if (moduleName[0] === '.') {
```

Comment:
.startsWith(0)? Whatever is faster on latest V8

Reply:
Totally forgot that function exists, would be cleaner for sure.
Not sure if performance is that critical here (how often would a user call `require.resolve.paths`?) but I can try to find/run a benchmark later.

Reply:
Okay so if you run it a lot, `[0] === '.'` is many times faster than `.startsWith('.')`.
I think because of this and because the `startsWith` only really simplifies the code a lot when called with a multi character string, we can leave it this way.

Reply:
When did V8 define string access via bracket notation? If it works on Node 6, then we're fine with it.

Reply:
CI definitely passed on Node 6 last time.
I also think I've seen this being done somewhere else in the codebase already, let me check.

Reply:
[Here](https://github.com/facebook/jest/blame/81d6888a971f9ff92a2e11f8fdc699add7c0f58d/packages/jest-resolve/src/index.js#L214) from a few years ago, so this should not cause any Node version issues.